### ISSUE TITLE
Route precedence 

### ISSUE THREAD

**Issue by marcusbooyah** [aspnet/Blazor#1042]
I have an app with two routes:

`@page "/{multiple}/catalog/new"`
`@page "/{multiple}/catalog/{catalogId}"`

When accessing the /{multiple}/catalog/new URL, the router seems to be taking me to the /{multiple}/catalog/{catalogId} route. This used to work the other way around for me. Is this a bug, or am I doing something wrong?

**Comment by Andrzej-W** [comment]
First route should have priority. Read comment at line 40 here
https://github.com/aspnet/Blazor/blob/dev/src/Microsoft.AspNetCore.Blazor/Routing/RouteTable.cs
Do you have these `@page` directives in the same `cshtml` file or in two different files?

**Comment by marcusbooyah** [comment]
Yeah, that's what I expected the behavior to be. It used to work this way for me. I don't recall when exactly it stopped working.

The `@page` directives are in two separate files.

**Comment by marcusbooyah** [comment]
For some reason, my `.csproj` had changed for this page only. Not sure if there was a bug that maybe caused this?

It contained this:
  ```
 <ItemGroup>
    <Content Remove="Pages\NewProduct.cshtml" />
  </ItemGroup>
```
which should be this:
```
 <ItemGroup>
    <BlazorGenerate Remove="Pages\NewProduct.cshtml" />
  </ItemGroup>
```

I'm now back up and running, but I'll leave this open for visibility in case anyone else has this problem.

Thanks!
